src/csync/csync_exclude.cpp

```diff
@@ -326,8 +326,9 @@ bool ExcludedFiles::loadExcludeFile(const QByteArray & basePath, const QString &
     }
 
     // nothing to prepare if the user decided to not exclude anything
-    if(_allExcludes.size())
+    if(_allExcludes[basePath].size()){
```

Comment:
Shouldn't that be `if (_allExcludes.contains(basePath)) {` ?

Sidenote: square bracket operator is generally not the best idea in contexts like this, since it inserts default constructed values for missing keys, growing the map for nothing.

Reply:
> since it inserts default constructed values for missing keys

is that why with _allExcludes.contains it never returns true now? 

so I tried it and now the exclude list doesn't load and it doesn't exclude any file...
I will need to work more on it then.

I added the space after the if.

Reply:
Ah... other uses of [] somewhere?

Reply:
Note that maybe you could change the `[]` operator for `value()`. This at least has the advantage of not inserting the default constructed value if the key is missing. Also I wouldn't use `size()` but `!...isEmpty()`